tests/unit_tests/shared/test_rate_keys.py

```diff
@@ -0,0 +1,60 @@
+import hypothesis.strategies as st
+from hypothesis import given
+from resdata.summary import Summary
+
+from ert.shared.storage.summary_key_utils import is_rate
+from tests.unit_tests.config.summary_generator import summary_variables
+
+
+def nonempty_string_without_whitespace():
+    return st.text(
+        st.characters(whitelist_categories=("Lu", "Ll", "Nd", "P")), min_size=1
+    )
+
+
+@given(key=nonempty_string_without_whitespace())
+def test_is_rate_does_not_raise_error(key):
+    is_rate_bool = is_rate(key)
+    assert isinstance(is_rate_bool, bool)
+
+
+examples = [
+    ("OPR", False),
+    ("WOPR:OP_4", True),
+    ("WGIR", True),
+    ("FOPT", False),
+    ("GGPT", False),
+    ("RWPT", False),
+    ("COPR", True),
+    ("LPR", False),
+    ("LWPR", False),
+    ("LCOPR", True),
+    ("RWGIR", True),
+    ("RTPR", True),
+    ("RXFR", True),
+    ("XXX", False),
+    ("YYYY", False),
+    ("ZZT", False),
+    ("SGPR", False),
+    ("AAPR", False),
+    ("JOPR", False),
+    ("ROPRT", True),
+    ("RNFT", False),
+    ("RFR", False),
+    ("RRFRT", True),
+    ("ROC", False),
+    ("BPR:123", False),
+    ("FWIR", True),
+]
+
+
+@given(example=st.sampled_from(examples))
```

Comment:
Why not just loop over the examples using pytest.mark.parametrize? I guess st.sampled_from will be happy if just a subset passes the test examples.

Reply:
Hm, I thought that this goes over all the examples anyway :smile:

Reply:
Done

Reply:
Maybe it does, but not even the documentation clearly states that it will always test all cases.